Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm

```diff
@@ -10749,16 +10749,20 @@ - (void)dragInteraction:(UIDragInteraction *)interaction item:(UIDragItem *)item
 {
     _isAnimatingDragCancel = YES;
     RELEASE_LOG(DragAndDrop, "Drag interaction willAnimateCancelWithAnimator");
-    auto previewView = _dragDropInteractionState.takePreviewViewForDragCancel();
-    [previewView setAlpha:0];
-    [animator addCompletion:[protectedSelf = retainPtr(self), previewView = WTFMove(previewView), page = _page] (UIViewAnimatingPosition finalPosition) {
+    auto previewViews = _dragDropInteractionState.takePreviewViewsForDragCancel();
+    for (auto& previewView : previewViews)
+        [previewView setAlpha:0];
+
+    [animator addCompletion:[protectedSelf = retainPtr(self), previewViews = WTFMove(previewViews), page = _page] (UIViewAnimatingPosition finalPosition) {
         RELEASE_LOG(DragAndDrop, "Drag interaction willAnimateCancelWithAnimator (animation completion block fired)");
-        [previewView setAlpha:1];
-        page->dragCancelled();
-        page->callAfterNextPresentationUpdate([previewView = WTFMove(previewView), protectedSelf = WTFMove(protectedSelf)] {
-            [previewView removeFromSuperview];
-            protectedSelf->_isAnimatingDragCancel = NO;
-        });
+        for (auto& previewView : previewViews) {
+            [previewView setAlpha:1];
+            page->dragCancelled();
+            page->callAfterNextPresentationUpdate([previewView = WTFMove(previewView), protectedSelf = WTFMove(protectedSelf)] {
+                [previewView removeFromSuperview];
+                protectedSelf->_isAnimatingDragCancel = NO;
+            });
```

Comment:
You should loop through `previewViews` inside this callback, rather than call this on every iteration.